apps/froniusmonitor/fronius_monitor.star

```diff
@@ -0,0 +1,182 @@
+"""
+
+load("cache.star", "cache")
+load("encoding/base64.star", "base64")
+load("encoding/json.star", "json")
+load("http.star", "http")
+load("humanize.star", "humanize")
+load("render.star", "render")
+load("schema.star", "schema")
+
+URL = "http://{}/solar_api/v1/GetPowerFlowRealtimeData.fcgi"
+CACHE_TTL = 60
+DUMMY_DATA = {
+    "Body": {
+        "Data": {
+            "Site": {
+                "P_Grid": 144.20,
+                "P_Load": 900.54,
+                "P_PV": 756.34,
+            },
+        },
+    },
+}
+
+SOLAR_PANEL = base64.decode("""
+iVBORw0KGgoAAAANSUhEUgAAABUAAAAQCAYAAAD52jQlAAAArElEQVQ4ja2T2xGFIAxETxyrcOhM
+y7M0xzb2fjA4kYcPrvsFCSzhBExCZDLDAHwuxZ5ovEq+MfIaWkYSqt3ikdLmlkG38dcmx1U9v2h8
+721mVeaNRgkwpjCzbytTWABO43i4VDMe44lYqlaS4sb5ttKWiu5faQoL+7ae5pLKd+4ntQVPlCMo
+mHpmOcNWLGc7+ES+uFevmELJNcU8ugG+rRKOx9/XoKph40P8rR8wcGBXI4UlEQAAAABJRU5ErkJg
+gg==
+""")
+SOLAR_PANEL_OFF = base64.decode("""
+iVBORw0KGgoAAAANSUhEUgAAABUAAAAQCAYAAAD52jQlAAAAd0lEQVQ4je2TwQrAIAxDo+y7e+iP
+d5cJtY0iustgOWklsb4i8OsTKqxoZrZkLoX6r5FBRAAAqkrX7XIWXFmX3rijFDqTiEBVuz1D1bW+
+yjKFBASJqX96ZDiqRbbVH5yyTKGrilxbzaOrwLtdAs+gdgdEAwcf4lg3XnREWPIOZLAAAAAASUVO
+RK5CYII=
+""")
+GRID = base64.decode("""
+iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAt0lEQVQ4jYWQuxEDIQxEhcfFeIZm
+HEEtTimHTmiBa4KYZC+xGKEToITR5+0KEYkAADIi54ycM6z+2wIkSEQUQnArg4cAQxJk+LouAgDn
+nNPcULdcaq2Q/VrrPNN7BwDI1zIopTzvwMN6Ay2y2/A4oGsyf5lqiyil7N13OcPmHU4DMk/Rj/7v
++8E0wCJaoLWGFD1S9OB8wKcNJMiuE7z6swanlXcwg7puwlJAO8pDLWG5rlXfgv+4AXBeHhx5xCS7
+AAAAAElFTkSuQmCC
+""")
+HOUSE = base64.decode("""
+iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAb0lEQVQ4jcWQQQrAMAgE19KX9dL/
+v2h7arDRFT1VCIHEGVYBUbxu+ntUCybnkgBPJDu83jsSBbckGUyC7yklOnYUaEkSWwlcPwHQPGxm
+ls2fJQj9anmVAADOTlOVLhXsQJXuUB/d+l/w2UE1q/p7AIUnlBV3qmXkAAAAAElFTkSuQmCC
+""")
+DOTS_LTR = base64.decode("""
+R0lGODlhCgAFAIABAA31VAAAACH/C05FVFNDQVBFMi4wAwEAAAAh+QQJDwABACwAAAAACgAFAAAC
+CYyPmWAc7pRMBQAh+QQJDwABACwAAAAACgAFAAACCIyPqWAcrmIsACH5BAkPAAEALAAAAAAKAAUA
+AAIIjI+pAda8oioAOw==
+""")
+DOTS_RTL = base64.decode("""
+R0lGODlhCgAFAIABAP/RGwAAACH/C05FVFNDQVBFMi4wAwEAAAAh+QQJDwABACwAAAAACgAFAAAC
+CIyPqQHWvKIqACH5BAkPAAEALAAAAAAKAAUAAAIIjI+pYByuYiwAIfkECQ8AAQAsAAAAAAoABQAA
+AgmMj5lgHO6UTAUAOw==
+""")
+
+def main(config):
+    inverter_ip = config.str("inverter_ip")
+
+    if inverter_ip and inverter_ip != "":
+        url = URL.format(inverter_ip)
+        data = cache.get(url)
+        if not data:
+            rep = http.get(url, params = {})
+            if rep.status_code != 200:
+                fail("Fronius API request failed with status {}".format(rep.status_code))
+            data = rep.body()
+            cache.set(url, data, ttl_seconds = CACHE_TTL)
+        o = json.decode(data)
+    else:
+        o = DUMMY_DATA
+
+    data = o["Body"]["Data"]["Site"]
+    pv = data["P_PV"]
+    power_load = data["P_Load"]
+    grid = data["P_Grid"]
+
+    points = []
+    flows = []
+
+    img = SOLAR_PANEL_OFF if pv == None or pv == 0 else SOLAR_PANEL
```

Comment:
is this a Copilot thing? would have written 
```
img = SOLAR_PANEL if pv else SOLAR_PANEL_OFF
```

Reply:
I was a bit confused about this.

Fronius docs say:
> #this value is null if inverter is not running
<img width="586" alt="image" src="https://github.com/tidbyt/community/assets/29188449/da156c11-9610-4ec2-8544-14397b68d71c">

But on my own inverter, at night with no generation, it was returning 0 values.

`if pv` would be `True` if `0`. So I felt it reasonable to handle both `None` and `0`